planner/core/planbuilder.go

```diff
@@ -3474,8 +3474,15 @@ func (b *PlanBuilder) buildDDL(ctx context.Context, node ast.DDLNode) (Plan, err
 		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.IndexPriv, v.Table.Schema.L,
 			v.Table.Name.L, "", authErr)
 	case *ast.CreateTableStmt:
-		if v.TemporaryKeyword != ast.TemporaryNone && v.ReferTable != nil {
-			return nil, ErrOptOnTemporaryTable.GenWithStackByArgs("create table like")
+		if v.TemporaryKeyword != ast.TemporaryNone {
+			for _, cons := range v.Constraints {
+				if cons.Tp == ast.ConstraintForeignKey {
+					return nil, infoschema.ErrCannotAddForeign
+				}
+			}
+			if v.ReferTable != nil {
+				return nil, ErrOptOnTemporaryTable.GenWithStackByArgs("create table like")
+			}
```

Comment:
Remove this. We're capable of supporting it. Please file another PR to solve `create table like` issue.